confdgnmi/src/confd_gnmi_api_adapter.py

```diff
@@ -59,32 +63,183 @@ def get_inst(cls) -> GnmiServerAdapter:
 
     class SubscriptionHandler(GnmiServerAdapter.SubscriptionHandler):
 
-        def make_subscription_response(self) -> gnmi_pb2.SubscribeResponse:
+        def __init__(self, adapter, subscription_list):
+            super().__init__(adapter, subscription_list)
+            # TODO reuse with demo adapter?
+            self.monitored_paths = []
+            self.change_db = []
+            self.change_db_lock = threading.Lock()
+            self.change_thread = None
+            self.stop_pipe = None
+
+        def get_monitored_changes(self) -> []:
+            # TODO reuse with demo adapter ?
+            self.change_db_lock.acquire()
```

Comment:
You should use `with self.change_db_lock:` instead, to deal with exceptions.
Similarly, sockets' closing should better be done using a context block, whenever possible (such as in `get_maapi_save_string`.

Reply:
chaged to `with`